### PR TITLE
feat: universal meta resolving

### PR DESCRIPTION
**Pull request by pi0**
Previously we was depending on providers to resolve image meta (getting aspect ratio and also base64) this was adding complexity and sometimes extera requests. With this PR, using [nuxt-contrib/image-meta](https://github.com/nuxt-contrib/image-meta) (for Node.js) and `Image` + `Canvas` (for Browser), we can universally have ratio. (Future release of `image-meta` may eventually support Node.js if we can remove Buffer dependency)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@ function loadProvider (key: string, provider: any) {
 function handleStaticGeneration (nuxt: any) {
   const staticImages = {} // url ~> hash
 
-  nuxt.hook('vue-renderer:ssr:prepareContext', (renderContext) => {
+  nuxt.hook('vue-renderer:ssr:prepareContext', async (renderContext) => {
```
Comment (pi0) [inline src/index.ts:105]: Unwanted change?

Comment (farnabaz) [reply src/index.ts:105]: yes

**Review by farnabaz — APPROVED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -117,6 +123,7 @@ function handleStaticGeneration (nuxt: any) {
 
   nuxt.hook('generate:done', async () => {
     const { port } = nuxt.server.listeners[0]
+    console.log("345678", nuxt.server.listeners[0])
```
Comment (pi0) [inline src/index.ts]: ??

File: package.json [hunk 3]
```diff
@@ -27,6 +27,7 @@
     "hasha": "^5.2.2",
     "image-meta": "^0.0.1",
     "ipx": "0.4.0-rc.1",
+    "make-fetch-happen": "^8.0.10",
```
Comment (pi0) [inline package.json:30]: Are we using this package?

**Pull request merged**